### PR TITLE
feat: drand quicknet: allow scheduling drand quicknet upgrade before nv22 on 2k devnet

### PR DESCRIPTION
**Pull request by hanabi1224**
## Related Issues

Currently, there's code using the blockchain network version to determine using chained or unchained drand beacon verification. This PR tries to decouple the logic so that on 2k devnet the drand quicknet upgrade can be scheduled before the Dragon upgrade, to facilitate testing on forest side. The change has been verified by the forest devnet tests in https://github.com/ChainSafe/forest/pull/3969

## Proposed Changes

- decouple drand and blockchain network upgrade logic
- allow overriding `GenesisNetworkVersion` with an environment variable on devnet
- allow overriding `UpgradePhoenixHeight` with an environment variable on devnet
- (tests are done in https://github.com/ChainSafe/forest/pull/3969)

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [x] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [x] New features have usage guidelines and / or documentation updates in
  - [x] [Lotus Documentation](https://lotus.filecoin.io)
  - [x] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [x] Tests exist for new functionality or change in behavior
- [x] CI is green

### CODE REVIEW

**Review by arajasek — COMMENTED** [review]
This looks generally good to me, thanks a lot for the contribution! I have one suggested improvement, but otherwise this is landable.

File: chain/beacon/drand/drand.go [hunk 1]
```diff
@@ -37,7 +37,8 @@ var log = logging.Logger("drand")
 //
 // The root trust for the Drand chain is configured from build.DrandChain.
 type DrandBeacon struct {
-	client dclient.Client
+	network dtypes.DrandEnum
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply chain/beacon/drand/drand.go]: Fixed.

File: chain/beacon/beacon.go [hunk 2]
```diff
@@ -110,10 +113,10 @@ func ValidateBlockValues(bSchedule Schedule, nv network.Version, h *types.BlockH
 }
 
 func BeaconEntriesForBlock(ctx context.Context, bSchedule Schedule, nv network.Version, epoch abi.ChainEpoch, parentEpoch abi.ChainEpoch, prev types.BeaconEntry) ([]types.BeaconEntry, error) {
-	// Before nv22 we had "chained" beacons, and so required two entries at a fork
-	if nv < network.Version22 {
+	// When we have "chained" beacons, two entries at a fork are required.
+	currBeacon := bSchedule.BeaconForEpoch(epoch)
+	if currBeacon.IsChained() {
```
Comment (arajasek) [inline chain/beacon/beacon.go:118]: Can we simplify this to: 

```
		parentBeacon := bSchedule.BeaconForEpoch(parentEpoch)
		currBeacon := bSchedule.BeaconForEpoch(epoch)
		if parentBeacon != currBeacon && currBeacon.IsChained() {
// fork logic
}
```

Then we can also drop line 140 (and just use currBeacon everywhere).

**Review by arajasek — APPROVED** [review]
Approved with one nit.

File: chain/beacon/beacon.go (hunk 2 above)
Comment (hanabi1224) [reply chain/beacon/beacon.go]: Fixed

**Pull request merged**